pkg/service/contract/schema.go

```diff
@@ -0,0 +1,111 @@
+package contract
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/getkin/kin-openapi/openapi3"
+	"go.keploy.io/server/v2/pkg/models"
+	yamlLib "gopkg.in/yaml.v3"
+)
+
+func validateSchema(openapi models.OpenAPI) error {
+	openapiYAML, err := yamlLib.Marshal(openapi)
+	if err != nil {
+		return err
+	}
+	// Validate using kin-openapi
+	loader := openapi3.NewLoader()
+	doc, err := loader.LoadFromData(openapiYAML)
+	if err != nil {
+		return err
+
+	}
+	// Validate the OpenAPI document
+	if err := doc.Validate(context.Background()); err != nil {
+		return err
+	}
+
+	fmt.Println("OpenAPI document is valid.")
+	return nil
+}
+
+// GetAllTestsSchema retrieves all the tests schema from the schema folder.
+func (s *contract) GetAllTestsSchema(ctx context.Context) (map[string]map[string]*models.OpenAPI, error) {
+	testsFolder := filepath.Join("./keploy", "schema", "tests")
+
+	s.openAPIDB.ChangePath(testsFolder)
+	testSetIDs, err := os.ReadDir(testsFolder)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read tests directory: %w", err)
+	}
+
+	testsMapping := make(map[string]map[string]*models.OpenAPI)
+	for _, testSetID := range testSetIDs {
+		if !testSetID.IsDir() {
+			continue
+		}
+
+		tests, err := s.openAPIDB.GetTestCasesSchema(ctx, testSetID.Name(), "")
+		if err != nil {
+			return nil, fmt.Errorf("failed to get test cases for testSetID %s: %w", testSetID.Name(), err)
+		}
+
+		testsMapping[testSetID.Name()] = make(map[string]*models.OpenAPI)
+		for _, test := range tests {
+			testsMapping[testSetID.Name()][test.Info.Title] = test
+		}
+	}
+
+	return testsMapping, nil
+}
+
+func (s *contract) GetAllDownloadedMocksSchemas(ctx context.Context) ([]models.MockMapping, error) {
+	downloadMocksFolder := filepath.Join("./Download", "Mocks")
+
+	// Read the contents of the Download Mocks folder to get all service directories.
+	entries, err := os.ReadDir(downloadMocksFolder)
+	if err != nil {
+		// If there's an error reading the directory, return it.
+		return nil, fmt.Errorf("failed to read mocks directory: %w", err)
+	}
+	var mocksSchemasMapping []models.MockMapping
+	// Loop over each entry in the Download Mocks folder.
```

Comment:
See what part can be moved to DB layer because you are reading a lot of things here. Right now it's okay but it needs to be refactored, its better to add TODO.

Reply:
ok